back/src/storage/session.ts

```diff
@@ -31,6 +38,40 @@ export const addNewUser = (
   ];
 };
 
+export const saveRoomInfo = (newRoomInfo: RoomInfo) => {
+  const roomIndex = roomCollectionSession.findIndex(
+    (roomInfo: RoomInfo) => roomInfo.room === newRoomInfo.room
+  );
+
+  if (roomIndex !== -1) {
+    const nextRoomCollectionSession = produce(
+      roomCollectionSession,
+      (draftState: RoomInfo[]) => {
+        draftState[roomIndex].content += '\n' + newRoomInfo.content;
+      }
+    );
+    roomCollectionSession = nextRoomCollectionSession;
+  } else {
+    const nextRoomCollectionSession = produce(
+      roomCollectionSession,
+      (draftState: RoomInfo[]) => {
+        draftState.push(newRoomInfo);
+      }
+    );
+    roomCollectionSession = nextRoomCollectionSession;
+  }
+
+  console.log('roomcollectionsession', roomCollectionSession);
+};
```

Comment:
Let's apply some clean code here (same level of abstraction principle), something like

```tsx
export const saveRoomInfo = (newRoomInfo: RoomInfo) => {
  const roomIndex = getRoomIndexByName(newRoomInfo.room);

  roomCollectionSession =  (roomIndex === -1)  ?
                                  updateRoomContent(index, content)
                                 :
                                 insertRoom(newRoomInfo); 
}
```

The create the detail functions.

By following this code is easier to follow